### PR TITLE
Earn stETH

### PR DESCRIPTION
**Pull request by timongll**

### CODE REVIEW

File: webapp/src/components/Earn/EarnChart.tsx [hunk 1]
```diff
@@ -14,23 +20,66 @@ interface ProfitChartProps {
...
-  barrierPercentage,
+  lowerBarrierPercentage,
+  upperBarrierPercentage,
   moneynessRange,
   yieldRange,
+  vaultOption,
 }) => {
   const [hoveredIndex, setIndex] = useState<number>();
 
+  const optionMoneynessText = (
```
Comment (cjinghong) [inline webapp/src/components/Earn/EarnChart.tsx:46]: can be converted into useCallback

File: webapp/src/components/Earn/Payoff.tsx [hunk 2]
```diff
@@ -90,73 +100,80 @@ const Payoff: React.FC<ProfitCalculatorProps> = () => {
...
-      i <= Math.round(barrierPercentage * 100) * 100;
-      i += 1
-    ) {
-      array.push(
-        4 +
-          Math.abs(i / 100 / (barrierPercentage * 100)) *
-            (maxYield - baseYield) *
-            100
-      );
+  const maxYieldText = useCallback(() => {
```
Comment (cjinghong) [inline webapp/src/components/Earn/Payoff.tsx:144]: can make this a `useMemo`

File: webapp/src/components/Earn/Payoff.tsx [hunk 3]
```diff
@@ -166,7 +183,7 @@ const Payoff: React.FC<ProfitCalculatorProps> = () => {
             <ParagraphText fontWeight={500}>Max Yield (APY) </ParagraphText>
             <TooltipExplanation
               title="MAX YIELD"
-              explanation="The max yield is defined as the max payout if the price of the underlying asset is at the barrier at expiry. The formula used to compute the max yield is as follows: BASE APY + (MAX PERF * 4 * PARTICIPATION RATE + 1)^(365 / 28) - 1"
+              explanation={maxYieldText()}
```
Comment (cjinghong) [inline webapp/src/components/Earn/Payoff.tsx:186]: then you dont need to call this function here

File: webapp/src/components/Earn/Payoff.tsx [hunk 4]
```diff
@@ -90,73 +100,80 @@ const Payoff: React.FC<ProfitCalculatorProps> = () => {
...
+        return (
+          commonText +
+          "BASE APY + (MAX PERF * PARTICIPATION RATE + 1)^(365 / 7) - 1"
+        );
     }
+  }, [vaultOption]);
 
-    for (let i = 0; i < 2000; i += 1) {
-      rightArray.push(4);
+  const expectedYieldText = useCallback(() => {
```
Comment (cjinghong) [inline webapp/src/components/Earn/Payoff.tsx:161]: same, can make this into useMemo

File: webapp/src/pages/DepositPage/EarnPage.tsx [hunk 5]
```diff
@@ -327,11 +330,10 @@ const EarnPage = () => {
   const { vaultAccounts } = useVaultAccounts(vaultVersion);
   const vaultAccount = vaultAccounts[vaultOption || VaultList[0]];
   const Logo = getAssetLogo(asset);
-
   const isLoading = status === "loading" || loading;
   useRedirectOnSwitchChain(getChainByVaultOption(vaultOption as VaultOptions));
 
-  let logo = <Logo height="100%" />;
+  const logo = <Logo height="100%" />;
```
Comment (cjinghong) [inline webapp/src/pages/DepositPage/EarnPage.tsx:336]: why is this stored in a variable instead of used directly?

**Review by cjinghong — COMMENTED** [review]
left comments

File: webapp/src/pages/DepositPage/EarnPage.tsx (hunk 5 above)
Comment (timongll) [reply webapp/src/pages/DepositPage/EarnPage.tsx:336]: it used is twiced so it was refactored

**Review by cjinghong — APPROVED** [review]

File: webapp/src/pages/DepositPage/EarnPage.tsx (hunk 5 above)
Comment (cjinghong) [reply webapp/src/pages/DepositPage/EarnPage.tsx:336]: gotcha

**Review by cjinghong — DISMISSED** [review]
g

**Review by leonkho — APPROVED** [review]

File: webapp/src/components/Vault/VaultActionsForm/EarnModal/ActionSteps.tsx [hunk 6]
```diff
@@ -67,6 +70,32 @@ const ActionSteps: React.FC<ActionStepsProps> = ({
...
+    return STEPS.formStep;
+  }, [v2WithdrawOption]);
+
+  const [minSTETHAmount, setMinSTETHAmount] = useState<BigNumber | undefined>();
+
+  const stETHDepositHelper = useSTETHDepositHelper(vaultVersion);
+
+  const { contract, getMinSTETHAmount } = useLidoCurvePool();
+
+  const [txhash, setTxhash] = useState("");
```
Comment (leonkho) [inline webapp/src/components/Vault/VaultActionsForm/EarnModal/ActionSteps.tsx:93]: Perhaps this should be kept consistent with the state above (`minSTETHAmount`):

`const [txHash, setTxHash] = useState<string | undefined>();`

**Review by leonkho — APPROVED** [review]

**Review by obadj3 — APPROVED** [review]
Lgtm

**Pull request merged**